recipes-devtools/rust/rust.inc

```diff
@@ -274,34 +274,46 @@ def arch_to_rust_target_arch(arch):
 
 # generates our target CPU value
 def llvm_cpu(d):
-    cpu = d.getVar('PACKAGE_ARCH')
-    target = d.getVar('TRANSLATED_TARGET_ARCH')
-
+    # First check if TUNE_CCARGS gives us a specific CPU to build for (via -march).
+    # Translate that GCC -march flag to a Rust/LLVM CPU.
+    trans = {}
+    trans['btver2'] = "btver2"
+    trans['core2'] = "core2"
+    trans['mips32'] = "mips32"
+    trans['mips32r2'] = "mips32r2"
+    trans['nehalem'] = "nehalem"
+    trans['skylake'] = "skylake"
+
+    for arg in (d.getVar('TUNE_CCARGS') or '').split():
+        if arg.startswith('-march='):
```

Comment:
I think `-march` is not universal, in some cases OE will use` -mcpu` e.g. ` -mcpu=power9` for ppc64 how does that get handled ?

Reply:
Currently, this is handled by not finding a `-march` parameter in `TUNE_CCARGS` and using the translation table for `TRANSLATED_TARGET_ARCH` instead, where "powerpc64" is translated to "ppc64".
If specific CPU optimization based on the `-mcpu` parameter becomes necessary in the future, someone should add another translation table and loop after the `-march` loop and translate `-mcpu` parameters to their Rust/LLVM equivalents there.

As the current PR doesn't break anything related to PowerPC, this can and should be done in a subsequent PR though.

Reply:
fair enough